### PR TITLE
INVS-880: Add cse Automation Service support

### PR DESCRIPTION
**Pull request by pmontiel-sumo**

### CODE REVIEW

**Comment by pmontiel-sumo** [comment]
Tested on local pointing to `csoar-stag-sandbox`, test passed.

```
GOROOT=/usr/local/opt/go/libexec #gosetup
GOPATH=/Users/pmontiel.ctr/go #gosetup
/usr/local/opt/go/libexec/bin/go test -c -o /private/var/folders/fp/79dmn6js7vvfdr18tlhmqk9r0000gp/T/GoLand/___TestAccSumologicSCEAutomation_create_update_in_github_com_SumoLogic_terraform_provider_sumologic_sumologic.test github.com/SumoLogic/terraform-provider-sumologic/sumologic #gosetup
/usr/local/opt/go/libexec/bin/go tool test2json -t /private/var/folders/fp/79dmn6js7vvfdr18tlhmqk9r0000gp/T/GoLand/___TestAccSumologicSCEAutomation_create_update_in_github_com_SumoLogic_terraform_provider_sumologic_sumologic.test -test.v -test.paniconexit0 -test.run ^\QTestAccSumologicSCEAutomation_create_update\E$
2023/01/05 11:47:07 Sumo Logic Terraform Provider Version=
=== RUN   TestAccSumologicSCEAutomation_create_update
2023/01/05 11:47:07 [DEBUG] Test: Executing step 0
2023/01/05 11:47:07 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:07 [INFO] terraform: building graph: GraphTypeValidate
2023/01/05 11:47:07 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:07 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodeValidatableResource) needs provider.sumologic
2023/01/05 11:47:07 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:07 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:07 [DEBUG] Starting graph walk: walkValidate
2023/01/05 11:47:07 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:07 [INFO] terraform: building graph: GraphTypeRefresh
2023/01/05 11:47:07 [DEBUG] Starting graph walk: walkRefresh
2023/01/05 11:47:07 [DEBUG] New state was assigned lineage "50e2b8ff-454e-d1e4-40c4-11b394bdd0c7"
2023/01/05 11:47:07 [INFO] terraform: building graph: GraphTypePlan
2023/01/05 11:47:07 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:07 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodePlannableResource) needs provider.sumologic
2023/01/05 11:47:07 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:07 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:07 [DEBUG] Starting graph walk: walkPlan
2023/01/05 11:47:07 [DEBUG] Resource instance state not found for node "sumologic_cse_automation.automation", instance sumologic_cse_automation.automation
2023/01/05 11:47:07 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:07 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:07 [WARN] Test: Step plan: DIFF:

CREATE: sumologic_cse_automation.automation
  cse_resource_type: "" => "INSIGHT"
  description:       "" => "<computed>"
  enabled:           "" => "true"
  execution_types.#: "" => "1"
  execution_types.0: "" => "NEW_INSIGHT"
  id:                "" => "<computed>"
  name:              "" => "<computed>"
  playbook_id:       "" => "638079aedb99cafada1e80a0"



STATE:

<no state>
2023/01/05 11:47:07 [INFO] terraform: building graph: GraphTypeApply
2023/01/05 11:47:07 [DEBUG] Resource state not found for node "sumologic_cse_automation.automation", instance sumologic_cse_automation.automation
2023/01/05 11:47:07 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation (prepare state)
2023/01/05 11:47:07 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation (prepare state)" (*terraform.NodeApplyableResource) needs provider.sumologic
2023/01/05 11:47:07 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodeApplyableResourceInstance) needs provider.sumologic
2023/01/05 11:47:07 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation (prepare state)" references: []
2023/01/05 11:47:07 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:07 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:07 [DEBUG] Starting graph walk: walkApply
2023/01/05 11:47:07 [DEBUG] sumologic_cse_automation.automation: applying the planned Create change
2023/01/05 11:47:11 [DEBUG] Request: [Method=POST] [URL=https://stag-api.sumologic.net/api/sec/v1/automations] [Headers=map[Authorization:[xxxxxxxxxxx] Content-Type:[application/json] User-Agent:[SumoLogicTerraformProvider/]]]. Response: [StatusCode=200 OK]
2023/01/05 11:47:11 [DEBUG] Request: [Method=GET] [URL=https://stag-api.sumologic.net/api/sec/v1/automations/0416235c-1d05-4cf5-8463-34de5fa3e863] [Headers=map[Authorization:[xxxxxxxxxxx] Content-Type:[application/json] User-Agent:[SumoLogicTerraformProvider/]]]. Response: [StatusCode=200 OK]
2023/01/05 11:47:11 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:11 [DEBUG] New state was assigned lineage "2370de45-22f1-5bbb-d07c-19e9b6ea46d1"
2023/01/05 11:47:12 [DEBUG] Request: [Method=GET] [URL=https://stag-api.sumologic.net/api/sec/v1/automations/0416235c-1d05-4cf5-8463-34de5fa3e863] [Headers=map[Authorization:[xxxxxxxxxxx] Content-Type:[application/json] User-Agent:[SumoLogicTerraformProvider/]]]. Response: [StatusCode=200 OK]
2023/01/05 11:47:12 [INFO] terraform: building graph: GraphTypePlan
2023/01/05 11:47:12 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:12 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodePlannableResource) needs provider.sumologic
2023/01/05 11:47:12 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:12 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:12 [DEBUG] Starting graph walk: walkPlan
2023/01/05 11:47:12 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:12 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:12 [INFO] terraform: building graph: GraphTypeRefresh
2023/01/05 11:47:12 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:12 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodeRefreshableManagedResource) needs provider.sumologic
2023/01/05 11:47:12 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:12 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:12 [DEBUG] Starting graph walk: walkRefresh
2023/01/05 11:47:12 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:13 [DEBUG] Request: [Method=GET] [URL=https://stag-api.sumologic.net/api/sec/v1/automations/0416235c-1d05-4cf5-8463-34de5fa3e863] [Headers=map[Authorization:[xxxxxxxxxxx] Content-Type:[application/json] User-Agent:[SumoLogicTerraformProvider/]]]. Response: [StatusCode=200 OK]
2023/01/05 11:47:13 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:13 [DEBUG] New state was assigned lineage "28124115-ed17-ea55-1319-01fafcaa2674"
2023/01/05 11:47:13 [INFO] terraform: building graph: GraphTypePlan
2023/01/05 11:47:13 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:13 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodePlannableResource) needs provider.sumologic
2023/01/05 11:47:13 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:13 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:13 [DEBUG] Starting graph walk: walkPlan
2023/01/05 11:47:13 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:13 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:13 [DEBUG] Test: Executing step 1
2023/01/05 11:47:13 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:13 [INFO] terraform: building graph: GraphTypeValidate
2023/01/05 11:47:13 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:13 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodeValidatableResource) needs provider.sumologic
2023/01/05 11:47:13 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:13 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:13 [DEBUG] Starting graph walk: walkValidate
2023/01/05 11:47:13 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:13 [INFO] terraform: building graph: GraphTypeRefresh
2023/01/05 11:47:13 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:13 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodeRefreshableManagedResource) needs provider.sumologic
2023/01/05 11:47:13 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:13 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:13 [DEBUG] Starting graph walk: walkRefresh
2023/01/05 11:47:13 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:14 [DEBUG] Request: [Method=GET] [URL=https://stag-api.sumologic.net/api/sec/v1/automations/0416235c-1d05-4cf5-8463-34de5fa3e863] [Headers=map[Authorization:[xxxxxxxxxxx] Content-Type:[application/json] User-Agent:[SumoLogicTerraformProvider/]]]. Response: [StatusCode=200 OK]
2023/01/05 11:47:14 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:14 [DEBUG] New state was assigned lineage "024c2d59-b937-46bd-f6bb-3a14e3eabb5f"
2023/01/05 11:47:14 [INFO] terraform: building graph: GraphTypePlan
2023/01/05 11:47:14 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:14 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodePlannableResource) needs provider.sumologic
2023/01/05 11:47:14 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:14 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:14 [DEBUG] Starting graph walk: walkPlan
2023/01/05 11:47:14 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:14 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:14 [WARN] Test: Step plan: DIFF:

UPDATE: sumologic_cse_automation.automation
  cse_resource_sub_types.#: "0" => "0"
  cse_resource_type:        "INSIGHT" => "INSIGHT"
  description:              "" => ""
  enabled:                  "true" => "false"
  execution_types.#:        "1" => "1"
  execution_types.0:        "NEW_INSIGHT" => "ON_DEMAND"
  id:                       "0416235c-1d05-4cf5-8463-34de5fa3e863" => "0416235c-1d05-4cf5-8463-34de5fa3e863"
  name:                     "QE General " => "QE General "
  playbook_id:              "638079aedb99cafada1e80a0" => "638079aedb99cafada1e80a0"



STATE:

sumologic_cse_automation.automation:
  ID = 0416235c-1d05-4cf5-8463-34de5fa3e863
  provider = provider.sumologic
  cse_resource_type = INSIGHT
  description = 
  enabled = true
  execution_types.# = 1
  execution_types.0 = NEW_INSIGHT
  name = QE General 
  playbook_id = 638079aedb99cafada1e80a0
2023/01/05 11:47:14 [INFO] terraform: building graph: GraphTypeApply
2023/01/05 11:47:14 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation (prepare state)
2023/01/05 11:47:14 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation (prepare state)" (*terraform.NodeApplyableResource) needs provider.sumologic
2023/01/05 11:47:14 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodeApplyableResourceInstance) needs provider.sumologic
2023/01/05 11:47:14 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation (prepare state)" references: []
2023/01/05 11:47:14 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:14 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:14 [DEBUG] Starting graph walk: walkApply
2023/01/05 11:47:14 [DEBUG] sumologic_cse_automation.automation: applying the planned Update change
2023/01/05 11:47:14 [DEBUG] Request: [Method=GET] [URL=https://stag-api.sumologic.net/api/sec/v1/automations/0416235c-1d05-4cf5-8463-34de5fa3e863] [Headers=map[Authorization:[xxxxxxxxxxx] Content-Type:[application/json] User-Agent:[SumoLogicTerraformProvider/]]]. Response: [StatusCode=200 OK]
2023/01/05 11:47:15 [DEBUG] Request: [Method=PUT] [URL=https://stag-api.sumologic.net/api/sec/v1/automations/0416235c-1d05-4cf5-8463-34de5fa3e863] [Headers=map[Authorization:[xxxxxxxxxxx] Content-Type:[application/json] If-Match:[] User-Agent:[SumoLogicTerraformProvider/]]]. Response: [StatusCode=200 OK]
2023/01/05 11:47:15 [DEBUG] Request: [Method=GET] [URL=https://stag-api.sumologic.net/api/sec/v1/automations/0416235c-1d05-4cf5-8463-34de5fa3e863] [Headers=map[Authorization:[xxxxxxxxxxx] Content-Type:[application/json] User-Agent:[SumoLogicTerraformProvider/]]]. Response: [StatusCode=200 OK]
2023/01/05 11:47:15 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:15 [DEBUG] New state was assigned lineage "b2350143-cfc8-a183-db47-6c452818d75a"
2023/01/05 11:47:16 [DEBUG] Request: [Method=GET] [URL=https://stag-api.sumologic.net/api/sec/v1/automations/0416235c-1d05-4cf5-8463-34de5fa3e863] [Headers=map[Authorization:[xxxxxxxxxxx] Content-Type:[application/json] User-Agent:[SumoLogicTerraformProvider/]]]. Response: [StatusCode=200 OK]
2023/01/05 11:47:16 [INFO] terraform: building graph: GraphTypePlan
2023/01/05 11:47:16 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:16 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodePlannableResource) needs provider.sumologic
2023/01/05 11:47:16 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:16 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:16 [DEBUG] Starting graph walk: walkPlan
2023/01/05 11:47:16 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:16 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:16 [INFO] terraform: building graph: GraphTypeRefresh
2023/01/05 11:47:16 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:16 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodeRefreshableManagedResource) needs provider.sumologic
2023/01/05 11:47:16 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:16 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:16 [DEBUG] Starting graph walk: walkRefresh
2023/01/05 11:47:16 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:17 [DEBUG] Request: [Method=GET] [URL=https://stag-api.sumologic.net/api/sec/v1/automations/0416235c-1d05-4cf5-8463-34de5fa3e863] [Headers=map[Authorization:[xxxxxxxxxxx] Content-Type:[application/json] User-Agent:[SumoLogicTerraformProvider/]]]. Response: [StatusCode=200 OK]
2023/01/05 11:47:17 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:17 [DEBUG] New state was assigned lineage "42e4e91e-678a-27a9-9ebb-b26a3b89b6b8"
2023/01/05 11:47:17 [INFO] terraform: building graph: GraphTypePlan
2023/01/05 11:47:17 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:17 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodePlannableResource) needs provider.sumologic
2023/01/05 11:47:17 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:17 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:17 [DEBUG] Starting graph walk: walkPlan
2023/01/05 11:47:17 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:17 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:17 [WARN] Test: Executing destroy step
2023/01/05 11:47:17 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:17 [INFO] terraform: building graph: GraphTypeValidate
2023/01/05 11:47:17 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:17 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodeValidatableResource) needs provider.sumologic
2023/01/05 11:47:17 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:17 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:17 [DEBUG] Starting graph walk: walkValidate
2023/01/05 11:47:17 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:17 [INFO] terraform: building graph: GraphTypeRefresh
2023/01/05 11:47:17 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:17 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodeRefreshableManagedResource) needs provider.sumologic
2023/01/05 11:47:17 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:17 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:17 [DEBUG] Starting graph walk: walkRefresh
2023/01/05 11:47:17 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:18 [DEBUG] Request: [Method=GET] [URL=https://stag-api.sumologic.net/api/sec/v1/automations/0416235c-1d05-4cf5-8463-34de5fa3e863] [Headers=map[Authorization:[xxxxxxxxxxx] Content-Type:[application/json] User-Agent:[SumoLogicTerraformProvider/]]]. Response: [StatusCode=200 OK]
2023/01/05 11:47:18 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:18 [DEBUG] New state was assigned lineage "1c8e7739-d1d0-f109-c8f6-09443ed65787"
2023/01/05 11:47:18 [INFO] terraform: building graph: GraphTypePlanDestroy
2023/01/05 11:47:18 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:18 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodePlanDestroyableResourceInstance) needs provider.sumologic
2023/01/05 11:47:18 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:18 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation (destroy)" (*terraform.NodeDestroyResourceInstance) needs provider.sumologic
2023/01/05 11:47:18 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodeAbstractResourceInstance) needs provider.sumologic
2023/01/05 11:47:18 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:18 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:18 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation (destroy)" references: []
2023/01/05 11:47:18 [DEBUG] Starting graph walk: walkPlanDestroy
2023/01/05 11:47:18 [WARN] Test: Step plan: DIFF:

DESTROY: sumologic_cse_automation.automation
  cse_resource_sub_types.#: "0" => ""
  cse_resource_type:        "INSIGHT" => ""
  description:              "" => ""
  enabled:                  "false" => ""
  execution_types.#:        "1" => ""
  execution_types.0:        "ON_DEMAND" => ""
  id:                       "0416235c-1d05-4cf5-8463-34de5fa3e863" => ""
  name:                     "QE General " => ""
  playbook_id:              "638079aedb99cafada1e80a0" => ""



STATE:

sumologic_cse_automation.automation:
  ID = 0416235c-1d05-4cf5-8463-34de5fa3e863
  provider = provider.sumologic
  cse_resource_type = INSIGHT
  description = 
  enabled = false
  execution_types.# = 1
  execution_types.0 = ON_DEMAND
  name = QE General 
  playbook_id = 638079aedb99cafada1e80a0
2023/01/05 11:47:18 [INFO] terraform: building graph: GraphTypeApply
2023/01/05 11:47:18 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation (prepare state)
2023/01/05 11:47:18 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation (destroy)" (*terraform.NodeDestroyResourceInstance) needs provider.sumologic
2023/01/05 11:47:18 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation (prepare state)" (*terraform.NodeApplyableResource) needs provider.sumologic
2023/01/05 11:47:18 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation (prepare state)" references: []
2023/01/05 11:47:18 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation (destroy)" references: []
2023/01/05 11:47:18 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:18 [DEBUG] adding implicit provider configuration provider.sumologic, implied first by sumologic_cse_automation.automation
2023/01/05 11:47:18 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation (destroy)" (*terraform.NodeDestroyResourceInstance) needs provider.sumologic
2023/01/05 11:47:18 [DEBUG] ProviderTransformer: "sumologic_cse_automation.automation" (*terraform.NodeAbstractResourceInstance) needs provider.sumologic
2023/01/05 11:47:18 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation" references: []
2023/01/05 11:47:18 [DEBUG] ReferenceTransformer: "sumologic_cse_automation.automation (destroy)" references: []
2023/01/05 11:47:18 [DEBUG] ReferenceTransformer: "provider.sumologic" references: []
2023/01/05 11:47:18 DestroyEdgeTransformer: pruning unused resource node sumologic_cse_automation.automation (prepare state)
2023/01/05 11:47:18 [DEBUG] Starting graph walk: walkDestroy
2023/01/05 11:47:18 [DEBUG] sumologic_cse_automation.automation: applying the planned Delete change
2023/01/05 11:47:19 [DEBUG] Request: [Method=DELETE] [URL=https://stag-api.sumologic.net/api/sec/v1/automations/0416235c-1d05-4cf5-8463-34de5fa3e863] [Headers=map[Authorization:[xxxxxxxxxxx] Content-Type:[application/json] User-Agent:[SumoLogicTerraformProvider/]]]. Response: [StatusCode=200 OK]
2023/01/05 11:47:19 [DEBUG] provider has no plugin.Client
2023/01/05 11:47:19 [DEBUG] New state was assigned lineage "2d9fc11f-809c-0b22-0f92-94a2f06984bf"
2023/01/05 11:47:19 [INFO] terraform: building graph: GraphTypePlanDestroy
2023/01/05 11:47:19 [DEBUG] Starting graph walk: walkPlanDestroy
2023/01/05 11:47:19 [INFO] terraform: building graph: GraphTypeRefresh
2023/01/05 11:47:19 [DEBUG] Starting graph walk: walkRefresh
2023/01/05 11:47:19 [DEBUG] New state was assigned lineage "bb784ef8-f40c-56fa-ec05-a29485402fb9"
2023/01/05 11:47:19 [INFO] terraform: building graph: GraphTypePlanDestroy
2023/01/05 11:47:19 [DEBUG] Starting graph walk: walkPlanDestroy
--- PASS: TestAccSumologicSCEAutomation_create_update (11.22s)
PASS

Process finished with the exit code 0

```

**Comment by pmontiel-sumo** [comment]
Matrix Test failing in other resources, but passed for this one:
```
=== RUN   TestAccSumologicSCEAutomation_create_update
--- PASS: TestAccSumologicSCEAutomation_create_update (19.85s)
```

File: sumologic/resource_sumologic_cse_automation_test.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	nExecutionTypes := []string{"NEW_INSIGHT"}
+	nEnabled := true
+	uExecutionTypes := []string{"ON_DEMAND"}
+	uEnabled := false
+	resourceName := "sumologic_cse_automation.automation"
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCSEAutomationDestroy,
+		Steps: []resource.TestStep{
```
Comment (mvbrock) [inline sumologic/resource_sumologic_cse_automation_test.go:26]: Should there be an update test in here as well? I see the two create ones.

**Review by mvbrock — APPROVED** [review]
Overall looks good, just left a couple questions.

File: sumologic/resource_sumologic_cse_automation_test.go (hunk 1 above)
Comment (pmontiel-sumo) [reply sumologic/resource_sumologic_cse_automation_test.go:26]: the test also does an update, see step:
```
			{
				Config: testCreateCSEAutomationConfig(nPlaybookId, nCseResourceType, uExecutionTypes, uEnabled),
				Check: resource.ComposeTestCheckFunc(
					testCheckCSEAutomationExists(resourceName, &Automation),
					testCheckAutomationValues(&Automation, nPlaybookId, nCseResourceType, uExecutionTypes, uEnabled),
				),
			},
```
it is updating the execution type and enabled field

File: sumologic/sumologic_cse_automation.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	return err
+}
+
+func (s *Client) CreateCSEAutomation(CSEAutomation CSEAutomation) (string, error) {
+
+	request := CSEAutomationRequestPost{
+		CSEAutomation: CSEAutomation,
+	}
+
+	var response CSEAutomationResponse
```
Comment (mvbrock) [inline sumologic/sumologic_cse_automation.go:39]: Minor nit, this could be right above line 46 for slightly improved clarity, my brain was like "why is this here?" until I read the next 5 lines.

File: sumologic/resource_sumologic_cse_automation.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+	CSEAutomation, err := c.GetCSEAutomation(id)
+	if err != nil {
+		log.Printf("[WARN] CSE Custom Entity Type not found when looking by id: %s, err: %v", id, err)
+
+	}
+
+	if CSEAutomation == nil {
+		log.Printf("[WARN] CSE Custom Entity Type not found, removing from state: %v - %v", id, err)
+		d.SetId("")
```
[the remark this answers is not in the document]
Comment (pmontiel-sumo) [reply sumologic/resource_sumologic_cse_automation.go:73]: if I remember correct, and based on the WARN I put in place is to remove from state, in case not found. Meaning the resource was deleted not using terraform ( via UI or API )

File: sumologic/sumologic_cse_automation.go (hunk 2 above)
Comment (mvbrock) [reply sumologic/sumologic_cse_automation.go:39]: e.g.
```
var response CSEAutomationResponse
err = json.Unmarshal(responseBody, &response)
```

File: sumologic/resource_sumologic_cse_automation_test.go (hunk 1 above)
Comment (mvbrock) [reply sumologic/resource_sumologic_cse_automation_test.go:26]: Oh, I see. I didn't read quite right to notice the difference between 'nExecutionTypes' and 'uExecutionTypes'.

File: sumologic/resource_sumologic_cse_automation.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mvbrock) [reply sumologic/resource_sumologic_cse_automation.go:73]: Oh I see that makes sense, you wouldn't want TF to hold onto it anymore.

**Review by josh-williams — APPROVED** [review]

**Comment by sumovishal** [comment]
Merging this PR. The failing tests are not related and tracked via separate Jira (SUMO-212855)
